libs/shared/contentful/src/lib/contentful.stripe-mapper.ts

```diff
@@ -0,0 +1,234 @@
+import { Stripe } from 'stripe';
+import {
+  ContentfulManager,
+  OfferingCommonContentResult,
+  PurchaseDetailsTransformed,
+} from '@fxa/shared/contentful';
+import { StripeMetadataKeys } from './types';
+
+/**
+ *  TypeGuard to ensure product on plan is a Product Object
+ */
+const isProductObject = (
+  product: Stripe.Plan['product']
+): product is Stripe.Product => {
+  if (typeof product !== 'string' && product && !product.deleted) {
+    return true;
+  } else {
+    return false;
+  }
+};
+
+/**
+ *  Class that maps Contentful Config onto a single Stripe Plan's metadata
+ */
+export class PlanMapper {
```

Comment:
Can we move this to it's own file?

Reply:
Do you think this should be in it's own library as well, or maybe in a sub folder? `shared-contentful` doesn't feel like the right home for these 2 classes.

Reply:
Ended up creating `payments-legacy` library. Let me know what you think.

Reply:
Liking the new legacy library approach, lets go with that.